protocol/sync.go

```diff
@@ -52,11 +52,9 @@ var syncCmd = &cobra.Command{
 			}
 		}
 
-		// TODO: state formatting
-		logger.Infof("Running sync with state: %v", state)
-
 		state.Mutex = &sync.Mutex{}
-
+		logger.Info("Running sync with state: ")
+		logger.LogState(state)
```

Comment:
it still not look good while printing 
can we do something like :
```suggestion
		state, _ := state.MarshalJSON()
		logger.Infof("Running sync with state: %s", state)
```

for this you have to change state unmarshaler as well. 

as that is not marking HoldsValue to true for each streams.

Reply:
it will print something like this 
```
{"type":"LOG","log":{"level":"info","message":"Running sync with state: {\"type\":\"STREAM\",\"streams\":[{\"stream\":\"incr1\",\"namespace\":\"incr\",\"sync_mode\":\"\",\"state\":{\"_data\":\"82678109C1000000022B0429296E1404\"}}]}"}}
```

Reply:
sure will make the changes accordingly

Reply:
@hash-data for this you have to change state unmarshaler as well. you mean marshaller ? sorry i'm having bit difficulty in understanding.

Reply:
is this what you are expecting ?
return json.Marshal(&struct {
		*Alias
		HoldsValue bool                   `json:"HoldsValue"`
		State      map[string]interface{} `json:"state"`
	}{
		Alias:      (*Alias)(s),
		State:      stateMap,
		HoldsValue: HoldsValue,
	})

Reply:
at the start you are going to unmarshal the state so, the holds value is not being added currently. So it needs to be added in Unmarshalling.

Reply:
okay now its clear, 
1)while loading we should mark holds value is going to be true based on incoming state map size,
2) while printing i.e marshalling do we need to print the holds value?

Reply:
nope, holds value use case is just to tell if stream state is present or not.

Reply:
got it we should load holds value based on incoming map size this is unmarshall function change.
while print we should print as you mentioned above adding both the changes.
Thanks

Reply:
pushed the changes pls review and lmk
{"type":"LOG","log":{"level":"info","message":"Running sync with state: {\"type\":\"STREAM\",\"streams\":[{\"stream\":\"stream_8\",\"namespace\":\"otter_db\",\"sync_mode\":\"cdc\",\"state\":{\"resume_token\":{\"_data\":\"82673F82FE000000022B0429296E1404\"}}},{\"stream\":\"stream_0\",\"namespace\":\"otter_db\",\"sync_mode\":\"cdc\",\"state\":{\"resume_token\":{\"_data\":\"82673F82FE000000022B0429296E1404\"}}}]}"}}